tests/sensors/test_python.py

```diff
@@ -41,6 +42,10 @@ def test_python_sensor_raise(self):
         with pytest.raises(ZeroDivisionError):
             self.run_as_task(lambda: 1 / 0)
 
+    def test_python_sensor_xcom(self):
+        poke_result =  self.run_as_operator(fn=lambda: PokeReturnValue(True, "xcom")).poke({})
```

Comment:
`self.run_as_operator` test class method use for execute operator and return task for further validations, see usage in:
- https://github.com/apache/airflow/blob/main/tests/operators/test_python.py

When you call `poke` on already created and executed task you actually call it again without provide actual context I would not recommend to do this because it could lead to potential issues in the future if something changes.

Otherwise I suggest two options (It is not limited by this options):
1. Run as a task and check resulting XCom value
2. Create task without helpers and manually call `poke` method

Reply:
I've opted to run the `poke` method without running the operator first. I wasn't sure how to properly get the context to check the xcom after the task is run.